packages/rum-core/src/domain/rumEventsCollection/view/trackViews.ts

```diff
@@ -262,9 +277,15 @@ function newView(
       lifeCycle.notify(LifeCycleEventType.VIEW_ENDED, { endClocks })
       clearInterval(keepAliveIntervalId)
       stopCommonViewMetricsTracking()
-      scheduleStopInitialViewMetricsTracking()
-      scheduleStopEventCountsTracking()
       triggerViewUpdate()
+      setTimeout(() => {
+        this.stop()
+      }, KEEP_TRACKING_AFTER_VIEW_DELAY)
+    },
+    stop() {
+      stopInitialViewMetricsTracking()
+      stopEventCountsTracking()
+      stopObservable.notify()
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review (if it exists :^)  -->
**👏 praise:** ‏love it